yt_dlp/extractor/slideslive.py

```diff
@@ -369,15 +398,47 @@ def _extract_custom_m3u8_info(self, m3u8_data):
 
         return m3u8_dict
 
-    def _extract_formats(self, cdn_hostname, path, video_id):
-        formats = []
-        formats.extend(self._extract_m3u8_formats(
+    def _parse_m3u8_vod_duration(self, m3u8_vod, video_id):
+        # SlidesLive VOD m3u8 does not contain '#EXT-X-PLAYLIST-TYPE:VOD'
+        if '#EXT-X-ENDLIST' not in m3u8_vod:
+            return None
+
+        return int(sum(
+            float(line[len('#EXTINF:'):].split(',')[0])
+            for line in m3u8_vod.splitlines() if line.startswith('#EXTINF:'))) or None
+
+    def _extract_mpd_vod_duration(
+            self, mpd_url, video_id, note=None, errnote=None, data=None, headers={}, query={}):
+
+        mpd_doc = self._download_xml(
+            mpd_url, video_id,
+            note='Downloading MPD VOD manifest' if note is None else note,
+            errnote='Failed to download VOD manifest' if errnote is None else errnote,
+            fatal=False, data=data, headers=headers, query=query) or {}
+        return int_or_none(parse_duration(mpd_doc.get('mediaPresentationDuration')))
```

Comment:
Is this useful enough to move to `extractor/common.py` to be with `_extract_m3u8_vod_duration`? If not, I can simplify it for this single use case

Reply:
Even if not useful now, it's general enough. You can move it to common